client/src/components/TaskCard/TaskCard.js

```diff
@@ -36,23 +42,38 @@ const TaskCard = ({
         }}
       />
     ) : (
-      <SmallActionButton text={"View"} toggleButton={showDetailsDialog} />
+      <SmallActionButton text={"View"} toggleButton={toggleModal} />
     );
 
+  const startDateTimeString = `${startDate} ${startTime}`;
+  const endDateTimeString = `${endDate} ${endTime}`;
+
   return (
     <div className={styles.taskContainer}>
+      {showModal ? (
+        <TaskDetails
+          closeModalHandler={toggleModal}
+          title={title}
+          location={location}
+          description={description}
+          startDateTimeString={startDateTimeString}
```

Comment:
We can probably omit the `String` part of the prop name here and just keep it as `startDateTime`

Reply:
Removed as indicated in comment below.